masp_primitives/src/transaction/components/sapling/builder.rs

```diff
@@ -160,6 +308,24 @@ impl<R: CryptoRng + RngCore> RngBuildParams<R> {
             outputs: BTreeMap::new(),
         }
     }
+
+    /// Convert these build parameters to their stored equivalent
+    pub fn to_stored(mut self) -> Option<StoredBuildParams> {
+        let mut stored = StoredBuildParams::default();
+        // Store the spends
+        for i in 0..self.spends.len() {
```

Comment:
can't you do 
```
Some(StoredBuildParams {
    spends: self.spends.into_values().collect(),
    convert_params: self.converts.into_values().collect(),
    output_params: self.outputs.into_values().collect(),
})
```

It should be slightly faster to as interators often optimize out bounds checking.

Reply:
Thanks. I was also trying to make sure that the map keys are consecutive numbers starting from 0 and return `None` otherwise. The bigger goal in this is to make sure that there are no missing randomness parameters during the conversion. Wouldn't the above fragment allow non-consecutive sequences of numbers in the map keys?

Reply:
Ah, I see. Indeed, what I proposed would allow non-consecutive sequences. Now I understand why you did this.